src/coreclr/pal/src/thread/process.cpp

```diff
@@ -2402,6 +2425,20 @@ PROCCreateCrashDump(std::vector<const char*>& argv)
             ERROR("PROCCreateCrashDump: prctl() FAILED %d (%s)\n", errno, strerror(errno));
         }
 #endif // HAVE_PRCTL_H && HAVE_PR_SET_PTRACER
+        close(child_pipe);
+
+        // Read createdump's stderr messages (if any)
+        if (errorMessageBuffer != nullptr && cbErrorMessageBuffer > 0)
+        {
+            // Read createdump's stderr
+            int count = read(parent_pipe, errorMessageBuffer, cbErrorMessageBuffer - 1);
+            if (count > 0)
+            {
+                errorMessageBuffer[count] = 0;
+            }
```

Comment:
I did have a question whether I needed to loop until count <= 0 on unix pipes.  I put it off until I was sure it was needed because it complicates the code quite a bit.

Reply:
something like this should work I think:
```c
int bytesRead = 0;
int count = 0;
while ((count = read(parent_pipe, errorMessageBuffer + bytesRead, cbErrorMessageBuffer - bytesRead) != 0)
{
    if (count < 0)
        // error

    bytesRead += count;
}
```